### ISSUE TITLE
Intermediate results should be logged

### ISSUE THREAD

**Issue by Tritlo** [Tritlo/Endemic#55]
Right now we wait until the very end to do any output, which is a shame if it found something already but then crashes or something. We should output patches to the output folder when they are found, to have some record at least.

**Comment by lapplislazuli** [comment]
Hmm, isn't this a bit of a scope creep? 

At the moment the (genetic) search does search and not much else, but then it would do patch-handling too. 

Also, I must say that at the moment endemic either fails at the beginning, or runs through 😅  I have not seen it failing somewhere in the middle.

**Comment by Tritlo** [comment]
True, but if I see a "1 success" now and `stop_on_results` is false, it keeps going. If I then do Ctrl-C, there's no way for me to see what that fix was 😢

**Comment by lapplislazuli** [comment]
I agree, I think it's worth it. 

May we put a delegate to pretty print in Utils and put it in all search algorithms? 
So like `printIntermediate :: Efix -> IO ()` in `Endemic.Search.Util` that is then called downstream? 

And have it maybe printed on `Verbose` or `Audit`.
